OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.java

```diff
@@ -123,7 +121,18 @@ public void onStop() {
         super.onStop();
         if (updateDownloadProgressRunnable != null) {
             handler.removeCallbacks(updateDownloadProgressRunnable);
-            wasDownloadItemVisibleBeforeStopping = downloadsMenuItem.isVisible();
+        }
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        if (updateDownloadProgressRunnable != null) {
+            handler.removeCallbacks(updateDownloadProgressRunnable);
+            /* Assign null on orientation change, and create new instance onCreateOptionsMenu that
+             *  updates the downloading progress of new attached progress view, else downloading
+             *  progress won't update.
```

Comment:
Following would be a better comment:

```java

 /* Assigning null here so that when this fragment is destroyed (e.g. due to orientation
  * change) the runnable is recreated and the download progress is updated properly.
  */
```